cmd/ncproxy/ncproxy.go

```diff
@@ -370,16 +478,22 @@ func (s *grpcService) DeleteNetwork(ctx context.Context, req *ncproxygrpc.Delete
 		return nil, status.Errorf(codes.InvalidArgument, "received empty field in request: %+v", req)
 	}
 
-	network, err := hcn.GetNetworkByName(req.Name)
-	if err != nil {
-		if _, ok := err.(hcn.NetworkNotFoundError); ok {
-			return nil, status.Errorf(codes.NotFound, "no network with name `%s` found", req.Name)
+	if _, err := s.ncproxyNetworking.GetNetworkByName(ctx, req.Name); err == nil {
+		if err := s.ncproxyNetworking.DeleteNetwork(ctx, req.Name); err != nil {
+			return nil, errors.Wrapf(err, "failed to delete network with name %q", req.Name)
+		}
+	} else {
+		network, err := hcn.GetNetworkByName(req.Name)
+		if err != nil {
+			if _, ok := err.(hcn.NetworkNotFoundError); ok {
```

Comment:
I don't believe they need to implement `As` to make use of it. This runs fine (hits the "Hello" branch):

```go
package main

import (
	"errors"
	"fmt"
	"log"

	"github.com/Microsoft/hcsshim/hcn"
)

func main() {
	_, err := hcn.GetNetworkByName("HCN_Network_That_100_Percent_Doesnt_Exist")
	if err != nil {
		var notfound hcn.NetworkNotFoundError
		if errors.As(err, &notfound) {
			fmt.Println("Hello")
		} else {
			log.Fatal(err)
		}
	}
}
```

Reply:
I'm not sure if they'd find the right type if they were wrapped with the github.com/pkg/errors package instead of the stdlib way of doing it though (the `fmt.Errorf("oh no: %w", err)` way)

Reply:
I didn't realize that worked; but changing error checking in ncproxy to use v1.13 syntax might make this PR even noiser, so I'm fine either way

Reply:
Yea I'm fine with an "improvement for later". I'd tried to replace all of our uses of github.com/pkg/errors over the holidays but we rely on the stack information that that package embeds in the errors in the gcs iirc

Reply:
I'm going to leave this as an improvement for later